netx/netx.go

```diff
@@ -175,13 +175,19 @@ func NewDialer(config Config) Dialer {
 	return d
 }
 
-// NewHTTP3Dialer creates a new DNS Dialer for HTTP3 transport, with the resolver from the specified config
-func NewHTTP3Dialer(config Config) HTTP3Dialer {
+// NewQUICDialer creates a new DNS Dialer for QUIC, with the resolver from the specified config
+func NewQUICDialer(config Config) QUICDialer {
 	if config.FullResolver == nil {
 		config.FullResolver = NewResolver(config)
 	}
-	d := &dialer.HTTP3DNSDialer{Resolver: config.FullResolver}
-	var dialer HTTP3Dialer = &httptransport.HTTP3WrapperDialer{Dialer: d}
+	var d dialer.QUICContextDialer = &dialer.QUICSystemDialer{Saver: config.ReadWriteSaver}
```

Comment:
I also mentioned above that I don't 100% understand this saver here. Maybe I'm missing something?

Reply:
Update: I think now I understand its purpose. Please refer to this comment: https://github.com/ooni/probe-engine/pull/1126#discussion_r553470896.

Reply:
Do you think the ReadWriteSaver should be encapsulated in a higher-level Dialer? In this case I'd have to provide access to the UDPconn struct to higher level Dialers.

Reply:
I'll ask you more details in chat. This part of the convo seems complex.

Reply:
So, for the records. That is the only place where we can observe a UDP connection and wrap it. Doing this kind of wrapping elsewhere is possible, but seems also less practical.